### ISSUE TITLE
Standardize on branch naming scheme?

### ISSUE THREAD

**Issue by sshah-wework** [MarquezProject/marquez#279]
Our branch naming patterns are all over the map. Let's settle on a single approach. Open to different proposals, have followed the approach below loosely, but would be willing to follow it strictly.

https://stackoverflow.com/a/6065944

**Comment by wslulciuc** [comment]
I agree, some general naming guidelines on branch names is a good idea. The linked approach is interesting, and it's the first time I've seen it, but maybe hard to adopt.

What are your thoughts on **`[issue-xxx]-[brief-description]`**?

**Example:**

```
issue-123-add-foo-class
```

**Comment by ashulmanWeWork** [comment]
I like that suggestion. Good for searching, and a brief reminder to avoid a lookup.

**Comment by sshah-wework** [comment]
I like the `[brief-description]`, the issues numbers alone is a little confusing for figuring out what's in flight.

**Comment by wslulciuc** [comment]
> the issues numbers alone is a little confusing for figuring out what's in flight

The alternative would be to use groups as pointed out in the linked stackoverflow?
 
```
$ git branch --list "*/foo"
new/foo
test/foo
ver/foo
```

We could adopt something like: **`[group]/[brief-description]`**, but what group list did you have in mind @sshah-wework?

**Comment by wslulciuc** [comment]
@sshah-wework : ping :)

**Comment by sshah-wework** [comment]
I like the `wip / feat / bug / junk` groups. if we remove issue numbers entirely, I'm ok with that too. the commit description should mention which issue is being addressed.

**Comment by wslulciuc** [comment]
I like the idea! In case a PR has been open, it would also be great to map a group to a label? I was thinking:

- `wip/iceberg-table-versioning` **=>** ["in progress"](https://github.com/MarquezProject/marquez/labels/in%20progress)
- `feat/list-dataset-versions` **=>** ["feature"](https://github.com/MarquezProject/marquez/labels/feature)
- `bug/dataset-version-query` **=>** ["bug"](https://github.com/MarquezProject/marquez/labels/bug)
- `exp/stream-versioning-support` **=>** ["experimental"](https://github.com/MarquezProject/marquez/labels/experimental)

If this sounds good, let's add a section on _Branch Naming_ to [CONTRIBUTING.md](https://github.com/MarquezProject/marquez/blob/master/CONTRIBUTING.md)

**Comment by ashulmanWeWork** [comment]
In favor of the groups idea :)

**Comment by sshah-wework** [comment]
I think the groups "feat", "bug" and "wip" make sense. Will add this to the contributing guide.
